test/crud-with-options.test.js

```diff
@@ -62,15 +88,49 @@ describe('crud-with-options', function () {
         });
       });
 
+    it('should allow findById(id, filter, cb) for a matching id',
+      function(done) {
+        User.create({name: 'x', email: 'x@y.com'}, function(err, u) {
+          should.not.exist(err);
+          should.exist(u.id);
+          User.findById(u.id, filter, function(err, u) {
+            should.exist(u);
+            should.not.exist(err);
+            u.should.be.an.instanceOf(User);
+            u.should.have.property('name', 'x');
+            u.should.have.property('email', undefined);
+            done();
+          });
+        });
+      });
```

Comment:
Can you please add a copy of this test using Promise-based API, to ensure that `findById(id, filter).then()` works correctly?